### PR TITLE
Update 'documents_columns' param passed to 'Sparse2Corpus'

### PR DESCRIPTION
**Pull request by chinmayapancholi13**
Fixes issue https://github.com/RaRe-Technologies/gensim/issues/1676.
cc: @mmunozm @menshikh-iv

### CODE REVIEW

**Comment by menshikh-iv** [comment]
I'm glad to see you @chinmayapancholi13, have you returned to us :smiley: ?

File: gensim/sklearn_api/ldamodel.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def fit(self, X, y=None):
         Calls gensim.models.LdaModel
         """
         if sparse.issparse(X):
-            corpus = matutils.Sparse2Corpus(X)
+            corpus = matutils.Sparse2Corpus(X, False)
```
Comment (piskvorky) [inline gensim/sklearn_api/ldamodel.py]: Named parameter better, for clarity. (here and elsewhere)

File: gensim/test/test_sklearn_api.py [hunk 2]
```diff
@@ -177,7 +177,7 @@ def testCSRMatrixConversion(self):
         newmodel.fit(sarr)
         bow = [(0, 1), (1, 2), (2, 0)]
         transformed_vec = newmodel.transform(bow)
-        expected_vec = numpy.array([0.35367903, 0.64632097])
+        expected_vec = numpy.array([0.12843782, 0.87156218])
```
Comment (piskvorky) [inline gensim/test/test_sklearn_api.py]: Why this change?

Comment (menshikh-iv) [reply gensim/test/test_sklearn_api.py]: I think because now this script interprets the input differently -> different model -> different result, but looks suspicious, I agree.

Comment (chinmayapancholi13) [reply gensim/test/test_sklearn_api.py]: @piskvorky As Ivan correctly pointed out, since the input matrix used for training (by `fit` method) is being interpreted differently after making this change, the model changes and thus, the output of `transform` changes as well.

File: gensim/sklearn_api/ldamodel.py (hunk 1 above)
Comment (chinmayapancholi13) [reply gensim/sklearn_api/ldamodel.py]: Thanks! Done now.

**Comment by chinmayapancholi13** [comment]
@menshikh-iv Right now, among the unit-tests, we have a function `testCSRMatrixConversion()` only for LDA model (and not for HDP and LSI models). So should I add tests with CSR matrices for these models as well?

And should we add separate unit-tests to ensure that input matrix is being inpterpreted in the right manner or just having a function like the current version of `testCSRMatrixConversion()` would be fine? I mean that currently `testCSRMatrixConversion()` function compares the output of `transform()` on the trained model (with a fixed expected output vector) after calling `fit()`. So we are inherently verifying that the input matrix is being interpreted in the right manner.

Also, I'm sorry for the delayed response. I am still a little busy in some university projects so it might take a few more weeks to resume the work here. :smile:

**Comment by menshikh-iv** [comment]
I dont think that additional tests is needed here (all changes exactly similar), LGTM for me, @piskvorky wdyt?

**Comment by piskvorky** [comment]
Up to you. Just checking: no default gensim behaviour was changed here, right? Only the wrapper.

**Comment by menshikh-iv** [comment]
@piskvorky you are correct (only wrapper).
@chinmayapancholi13 thanks for fix :+1:
